rails_docker/Dockerfile

```diff
@@ -36,8 +37,26 @@ RUN locale-gen en_US.UTF-8
 RUN mkdir "$APP_HOME"
 WORKDIR $APP_HOME
 
-COPY Gemfile* "$APP_HOME"/
+# Only copy the dependency definition files (Gemfile and packages) to use Docker cache for these steps
+# Install Ruby dependencies
+COPY Gemfile* ./
+COPY engines ./engines/
+RUN bundle install --jobs $BUNDLE_JOBS --path $BUNDLE_PATH
+
+# Install JS dependencies
+COPY package.json yarn.lock ./
+RUN yarn install
+
+# Copying the app files must be placed after the dependencies setup
+# since the app files always change thus cannot be cached
+COPY . ./
+
+# Compile assets
+RUN RAILS_ENV=$RAILS_ENV bundle exec rake assets:precompile
+
+# Make the init file executable
+RUN chmod +x ./bin/start.sh
 
-RUN bundle install
+EXPOSE $PORT
```

Comment:
@olivierobert how about managing all the ports from docker-compose.yml file.

Reply:
It's is already in `docker-compose.yml`. But I could add it for `docker-compose.test.yml` too. But the thing is that it would still be needed in the Dockerfile as not all providers have Docker Compose deployment e.g. Heroku only has Docker ddeployments, no support for Docker Compose. So it's safer to still have it in the Dockerfile.